src/main/java/entities/Tag.java

```diff
@@ -0,0 +1,42 @@
+package entities;
+
+import java.util.ArrayList;
+
+public class Tag {
+    // The name of the tag
+    final String name;
+
+    // Array of posts that are under this tag
+    final ArrayList<Post> taggedPosts;
```

Comment:
Could you make this a List instead of ArrayList just so we can keep is as general as possible.